datapoint/cast.go

```diff
@@ -0,0 +1,72 @@
+package datapoint
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// CastIntegerValue casts a signed integer to a datapoint Value
+func CastIntegerValue(value interface{}) (metricValue Value, err error) {
+	switch value.(type) {
```

Comment:
See https://tour.golang.org/methods/16 for a way that you can avoid having to do the extra cast.